PublicFolders/ValidateEXOPFDumpster.ps1

```diff
@@ -0,0 +1,423 @@
+    $PublicFolderInfo.NON_IPM_SUBTREE | Format-List | Out-File -FilePath "$ExportPath\logs_$tstamp\NON_IPM_SUBTREE.txt" -NoClobber
+    $PublicFolderInfo.DUMPSTER_ROOT | Format-List | Out-File -FilePath "$ExportPath\logs_$tstamp\DUMPSTER_ROOT.txt" -NoClobber
+    Compress-Archive -Path "$ExportPath\logs_$tstamp" -DestinationPath $ExportPath\logs_$tstamp
+}
+#Intro
+$ts = Get-Date -Format yyyyMMdd_HHmmss
+Write-Host $ExportPath
+If ($null -eq $ExportPath -or $ExportPath -eq "") {
+    $ExportPath = "$env:USERPROFILE\Desktop\ValidatePFDumpster\ValidatePFDumpster_$ts"
+    mkdir $ExportPath -Force | Out-Null
+} else {
+    if (!(Test-Path -Path $ExportPath)) {
+        Write-Host "The specified folder location $ExportPath is not existing, please re-run the script and ensure to enter a valid path or leave the ExportPath unassigned" -ForegroundColor Red
+        exit
+    } else {
+        $ExportPath = "$ExportPath\ValidatePFDumpster\ValidatePFDumpster_$ts"
+        mkdir $ExportPath -Force | Out-Null
+    }
+}
+$Script:Conditionsfailed = 0
+$Pfolder=Read-Host "Please enter affected public folder EntryID or Identity: e.g \PF1"
+[string]$Description = "This script illustrates issues related to deleting public folder items or removing the public folder as awhole on Publicfolder $Pfolder, BLOCKERS will be reported down, please ensure to mitigate them!`n"
+Write-Host $Description -ForegroundColor Cyan
+$Description | Out-File $ExportPath\ValidatePFDumspterREPORT.txt -Append
+
+#Connect to EXO PS
+$Sessioncheck = Get-PSSession | Where-Object { $_.Name -like "*Exchangeonline*" -and $_.State -match "opened" }
+if ($null -eq $Sessioncheck) {
+    Connect2EXO
+}
+
+#Main Function
+$PublicFolderInfo=GetPublicFolderInfo($Pfolder)
+ValidatePublicFolderIssue($PublicFolderInfo)
+ValidateContentMBXUniqueness($PublicFolderInfo)
+ValidateEntryIDMapping($PublicFolderInfo)
+ValidateContentMBXQuota($PublicFolderInfo)
+ValidatePublicFolderQuota($PublicFolderInfo)
+ValidateDumpsterFlag($PublicFolderInfo)
+ValidateDumpsterChildren($PublicFolderInfo)
+ValidateMEPFGuid($PublicFolderInfo)
+ValidateParentPublicFolder($PublicFolderInfo)
+
+
+#Ask for feedback
+Write-Host "Please rate the script experience & tell us what you liked or what we can do better over https://aka.ms/PFDumpsterFeedback!" -ForegroundColor Cyan
+"Please rate the script experience & tell us what you liked or what we can do better over https://aka.ms/PFDumpsterFeedback!" | Out-File $ExportPath\ValidatePFDumspterREPORT.txt -Append
+
+#Quit EXO session
+try {
+    Write-Host "Quiting EXO PowerShell session..." -ForegroundColor Yellow
+    Disconnect-ExchangeOnline -ErrorAction Stop -Confirm:$false
+    $CurrentDescription= "Disconnecting from EXO V2"
+    $CurrentStatus = "Success"
+    logerror -CurrentStatus $CurrentStatus -Function "Disconnecting from EXO V2" -CurrentDescription $CurrentDescription
```

Comment:
Why are you disconnecting from EXO session if the session was already created prior to running the script?

Reply:
Fixed that by quitting EXO session only when the script created the EXO session otherwise the session shall stay untouched.